processor/signozlogspipelineprocessor/stanza/operator/operators/preprocessing/transformer.go

```diff
@@ -0,0 +1,75 @@
+package json
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	signozstanzaentry "github.com/SigNoz/signoz-otel-collector/processor/signozlogspipelineprocessor/stanza/entry"
+	signozstanzahelper "github.com/SigNoz/signoz-otel-collector/processor/signozlogspipelineprocessor/stanza/operator/helper"
+	"github.com/SigNoz/signoz-otel-collector/utils"
+	"github.com/bytedance/sonic"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/entry"
+)
+
+var msgCompatibleFields = []string{"msg", "log", "log.message", "log.msg", "log.log"}
+
+type Processor struct {
+	signozstanzahelper.TransformerOperator
+}
+
+// Process will parse an entry for JSON.
+func (p *Processor) Process(ctx context.Context, entry *entry.Entry) error {
+	return p.TransformerOperator.ProcessWith(ctx, entry, p.transform)
+}
+
+func (p *Processor) ProcessBatch(ctx context.Context, entries []*entry.Entry) error {
+	return p.ProcessBatchWith(ctx, entries, p.Process)
+}
+
+// preprocess log body
+func (p *Processor) transform(entry *entry.Entry) error {
+	parsedValue := make(map[string]any)
+	switch v := entry.Body.(type) {
+	case string:
+		// Unquote JSON strings if possible
+		err := sonic.Unmarshal([]byte(utils.Unquote(v)), &parsedValue)
```

Comment:
should we do some basic precheck before unmarshal eg: if it doesn't start with {, no point in trying to unmarshal it.